src/mlpack/core/util/prefixedoutstream.hpp

```diff
@@ -124,11 +124,19 @@ class PrefixedOutStream
    * Conducts the base logic required in all the operator << overloads.  Mostly
    * just a good idea to reduce copy-pasta.
    *
+   * Calls 2 different overloads to be able to output arma objects with custom
+   * precision
+   *
    * @tparam T The type of the data to output.
    * @param val The The data to be output.
    */
   template<typename T>
-  void BaseLogic(const T& val);
+  typename std::enable_if<!arma::is_arma_type<T>::value, void>::type
```

Comment:
No need for the second template parameter to `enable_if`---it's already default `void`.